src/api.py

```
@@ -0,0 +1,390 @@
+    """Get detailed system status and statistics"""
+    try:
+        status = agent_news_app.get_system_status()
+        
+        return APIResponse(
+            success=True,
+            message="System status retrieved successfully",
+            data=status
+        )
+    except Exception as e:
+        logger.error(f"Error getting system status: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)